packages/web3-eth/src/utils/send_tx_helper.ts

```diff
@@ -122,7 +122,14 @@ export class SendTxHelper<
 
 	public async checkRevertBeforeSending(tx: TransactionCall) {
 		if (this.options.checkRevertBeforeSending !== false) {
-			const reason = await getRevertReason(this.web3Context, tx, this.options.contractAbi);
+			let formatTx = tx;
+			if (isNullish(tx.data) && isNullish(tx.input) && isNullish(tx.gas)) { // eth.call runs into error if data isnt filled and gas is not defined, its a simple transaction so we fill it with 21000
```

Comment:
when data is populated, is gas limit always populated in that case?

Reply:
and its only missing in case of value transfers?

Reply:
additionally, if tx has gas limit only for eth_call rpc( for revert check) and not in actual Tx that will be sent after it, 
eth_call may pass while actual Tx may fail in some cases, as both will differ, so I think gas limit should be applied in all cases